src/EventEmitterInterface.php

```diff
@@ -13,10 +13,38 @@
 
 interface EventEmitterInterface
 {
-    public function on($event, callable $listener);
-    public function once($event, callable $listener);
-    public function removeListener($event, callable $listener);
-    public function removeAllListeners($event = null);
-    public function listeners($event = null);
-    public function emit($event, array $arguments = []);
+    /**
+     * Allows you to subscribe to an event.
+     */
+    public function on(string $event, callable $listener): static;
+
+    /**
+     * Convenience method that adds a listener which is guaranteed to only be
+     * called once.
+     */
+    public function once(string $event, callable $listener): static;
+
+    /**
+     * Remove a specific listener for a specific event.
+     */
+    public function removeListener(string $event, callable $listener): void;
```

Comment:
IMHO it would be worth adding the specific `callable` signature we expect to the docblocks here.

Reply:
I'm open to ideas/suggestions. The only issue I see with adding a specific signature is that a listener can be any callable.

Reply:
As mentioned at https://github.com/igorw/evenement/pull/88#issuecomment-2205118391 this will be taken care of in a follow up PR